dv/verilator/sonata_system.cc

```diff
@@ -38,6 +38,17 @@ int SonataSystem::Main(int argc, char **argv) {
 int SonataSystem::Setup(int argc, char **argv, bool &exit_app) {
   VerilatorSimCtrl &simctrl = VerilatorSimCtrl::GetInstance();
 
+  // Default to CHERI enabled, but this may be overridden on the command line
+  // for running legacy non-CHERI software on the same build of the simulator.
+  _top.disable_cheri = 0;
+  for (int i = 1; i < argc; ++i) {
```

Comment:
Not really for this PR, but it's a bit sad to be reimplementing getopt here and then calling it later in `ParseCommandArgs`!

I wonder what the more sensible structure would be. Presumably this is a problem that gets solved all over the place. My (vague) memory of one solution has the first reader grab the arguments it cares about and then filter it out before passing the other arguments to the next reader. Either the last reader or the overarching code then checks that every argument was understood by *someone* :-)